syntax = "proto3";

package metrics;

message MetricRequest {
}

// metrics:
// - keys per minute (KPM)
// - avg mouse speed (pixel dist / min) (px/min)
// - clicks per minute (CPM)
// - active media/song titles every 30 seconds (multiple messages for multiple media)

service MetricService {
    rpc GetKPMStream(MetricRequest) returns (stream MetricResponse);
    rpc GetMouseSpeedStream(MetricRequest) returns (stream MetricResponse);
    rpc GetCPMStream(MetricRequest) returns (stream MetricResponse);
    rpc GetMediaStream(MetricRequest) returns (stream MetricResponse);
}

message MetricResponse {
    oneof data {
        int32 kpm = 1;
        float pxm = 2;
        float cpm = 3;
        string title = 4;
    }
    string timestamp = 5;
}